api/include/opentelemetry/trace/tracer.h

```diff
@@ -163,6 +163,18 @@ class Tracer
     }
   }
 
+#if OPENTELEMETRY_ABI_VERSION_NO >= 2
+  /**
+   * Reports if the tracer is enabled or not. A disabled tracer will not create spans.
+   *
+   * The instrumentation authors must call this method very time before creating a span to
+   * potentially avoid performing computationally expensive operations for disabled tracers.
+   *
+   * @since ABI_VERSION 2
+   */
+  virtual bool Enabled() noexcept = 0;
```

Comment:
Just wondering if need to do anything specific for this - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#enabled

-- clip
 Parameters can be added in the future, therefore, the API MUST be structured in a way for parameters to be added.


Adding new overload or modifying this method in future would break the ABIv2 compatibility, unless we are fine with that. cc @marcalff 
Else, one option could be to provide an overloaded method which accepts Context (similar to Logger::Enabled), which would provide a mechanism for passing additional parameters in the future.

But I am even fine to break the ABIv2 compatibility, as this API is still not stable in the specs.

Reply:
Re:

>  Parameters can be added in the future, therefore, the API MUST be structured in a way for parameters to be added.

This is wishful thinking from the spec, but ignores the reality of C++ constraints.

Both the spec for Enabled() and the ABI_VERSION_2 are not stable, this can be changed any time, and will be just fine.

Ok to merge as is (with a const)